### DISCUSSION TITLE
Force select a different device.

### DISCUSSION THREAD

**Discussion by iotinkognito** [make-all/tuya-local#901] [Show and tell]
Hi, I have some devices that are being incorrectly detected as Co2 devices even though they are GPO's. I modified the yaml for a device that was similar and changed the DP's accordingly and added the product_id. It still doesn't seem to pick it up via the ui integration. Is there a way to force it to use a particular devices\*.yaml? Or a way to select a modified devices yaml (i.e. add your own device). 

Thanks in advance

**Comment by make-all** [comment]
You need to get the log of dps that are actually coming from the device, and write a config yaml file to match that.

Any valid config file added to the devices directory will be picked up during device identification, but it will be overwritten if you upgrade the integration, so you should contribute it here to get it included (pull requests are dealt with quickest, but if you don't know how to do a GitHub pull request, a report with your config, logs and any other info you used to create the config will make checking it easy and push it towards the front of the queue.

**Reply by blakadder** [reply]
At the peril of hijacking the discussion, what are the requirements for the device to be autoidentified? I've had troubles getting my vacuums to autoidentify reliably when I had a full config but it worked with only a primary_entity configured. Adding product id didn't help

**Reply by make-all** [reply]
Any dps that might be missing need to be marked as optional.  As vacuums are quite complex devices they tend to have a lot of dps, and I guess to save memory on the module, some of them are only sent when the device is operating.
In extreme cases where there aren't enough dps consistently sent to identify the device, we might recommend that the device be switched on at least before trying to add it.
